kore/src/Kore/Step/Simplification/TermLike.hs

```diff
@@ -191,7 +191,7 @@ simplify sideCondition = \termLike ->
     simplifyInternalWorker termLike
     >>= ensureSimplifiedResult sideConditionRepresentation termLike
   where
-    sideConditionRepresentation = SideCondition.toRepresentation sideCondition
+    ~sideConditionRepresentation = SideCondition.toRepresentation sideCondition
```

Comment:
I have the same question here as in `Kore.Step.Simplification.SubstitutionSimplifier`.